File: build-an-email-masker.py
def mask_email(email):
    at_index = email.find("@")
    
    username = email[:at_index]
    domain = email[at_index:]
    
    first_char = username[0]
    last_char = username[-1]
    
    mask_length = len(username) - 2
    masked_middle = "*" * mask_length
    
    return f"{first_char}{masked_middle}{last_char}{domain}"

email = "myEmail@email.com"
print(mask_email(email))

File: build-an-email-masker.js
function maskEmail(email) {
  const atIndex = email.indexOf("@");
  
  const username = email.slice(0, atIndex);
  const domain = email.slice(atIndex);
  
  const firstChar = username[0];
  const lastChar = username[username.length - 1];
  
  const maskLength = username.length - 2;
  const maskedMiddle = "*".repeat(maskLength);
  
  return `${firstChar}${maskedMiddle}${lastChar}${domain}`;
}

const email = "myEmail@email.com";

console.log(maskEmail(email));